src/bios.rs

```diff
@@ -122,16 +122,24 @@ impl Component for Bios {
     }
 
     fn query_adopt(&self) -> Result<Option<Adoptable>> {
-        Ok(None)
+        crate::component::query_adopt_impl()
     }
 
-    #[allow(unused_variables)]
-    fn adopt_update(
-        &self,
-        sysroot: &openat::Dir,
-        update: &ContentMetadata,
-    ) -> Result<InstalledContent> {
-        todo!();
+    fn adopt_update(&self, _: &openat::Dir, update: &ContentMetadata) -> Result<InstalledContent> {
+        let meta = if let Some(meta) = self.query_adopt()? {
+            meta
+        } else {
+            anyhow::bail!("Failed to find adoptable system");
+        };
```

Comment:
Totally fine as is, we have a lot of this in the code but in newer Rust one can do instead:

`let Some(meta) = self.query_adopt()? else { anyhow::bail!("Failed to find adoptable system") };`
which is shorter. See e.g. https://github.com/containers/bootc/pull/650/files

Reply:
Will update this in later patch, thanks!